assembly/src/parsers/ast/mod.rs

```diff
@@ -0,0 +1,339 @@
+#![allow(dead_code)]
+use super::{AssemblyError, Token, TokenStream, Vec};
+use crate::MODULE_PATH_DELIM;
+use vm_core::utils::{collections::BTreeMap, string::String, string::ToString};
+
+mod nodes;
+use nodes::{Instruction, Node, Procedure};
+
+mod context;
+use context::ParserContext;
+
+mod io_ops;
+mod stack_ops;
+mod u32_ops;
+
+#[cfg(test)]
+pub mod tests;
+
+// TYPE ALIASES
+// ================================================================================================
+type ProcMap = BTreeMap<String, Procedure>;
+
+// Represents a parsed program AST.
+// This AST can then be furthur processed to generate MAST.
+// A program has to have a body and no exported procedures.
```

Comment:
nit: here and in other places in this file, these should be doc comments - i.e., `///`.

Reply:
fixed